broker/src/main/java/org/apache/rocketmq/broker/BrokerStartup.java

```diff
@@ -257,4 +268,28 @@ public static Options buildCommandlineOptions(final Options options) {
 
         return options;
     }
+
+    private static void checkMQAlreadyStart(String storePathRootDir) throws IOException {
+        File file = new File(StorePathConfigHelper.getLockFile(storePathRootDir));
+        MappedFile.ensureDirOK(file.getParent());
+
+        final RandomAccessFile randomAccessFile = new RandomAccessFile(file, "rw");
+        FileLock lock = randomAccessFile.getChannel().tryLock(0, 1, false);
+        if (lock == null || lock.isShared() || !lock.isValid()) {
+            throw new RuntimeException("Lock failed,MQ already started");
+        }
+
+        randomAccessFile.getChannel().write(ByteBuffer.wrap("lock".getBytes()));
+        randomAccessFile.getChannel().force(true);
+
+        Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
Also, release the file lock in `shutdown` of `MessageStore` is recommended.

Reply:
+1

Reply:
done